FirebaseStorageSwift/Sources/SwiftApiExtension.swift

```diff
@@ -0,0 +1,96 @@
+private func getResultCallback<T>(
+  completion: @escaping (Result<T, Error>) -> Void
+) -> (_: T?, _: Error?) -> Void {
+  return { (metadata: T?, error: Error?) -> Void in
+    guard let metadata = metadata else {
+      guard let error = error else {
+        completion(.failure(NSError(domain: "FirebaseStorageSwift",
+                                    code: -1,
+                                    userInfo: ["Storage Result Generator":
+                                      "InternalError - Return type and Error code both nil"])))
+        return
+      }
+      completion(.failure(error))
+      return
+    }
+    completion(.success(metadata))
+  }
+}
+
+public extension StorageReference {
+  func downloadURL(completion: @escaping (Result<URL, Error>) -> Void) {
+    downloadURL(completion: getResultCallback(completion: completion))
+  }
+
+  func getData(maxSize: Int64, completion: @escaping (Result<Data, Error>) -> Void)
+    -> StorageDownloadTask {
+    return getData(maxSize: maxSize, completion: getResultCallback(completion: completion))
+  }
+
+  func getMetadata(completion: @escaping (Result<StorageMetadata, Error>) -> Void) {
+    getMetadata(completion: getResultCallback(completion: completion))
+  }
+
+  func list(withMaxResults maxResults: Int64,
+            pageToken: String,
+            completion: @escaping (Result<StorageListResult, Error>) -> Void) {
+    list(withMaxResults: maxResults,
+         pageToken: pageToken,
+         completion: getResultCallback(completion: completion))
+  }
+
+  func list(withMaxResults maxResults: Int64,
+            completion: @escaping (Result<StorageListResult, Error>) -> Void) {
+    list(withMaxResults: maxResults,
+         completion: getResultCallback(completion: completion))
+  }
+
+  func listAll(completion: @escaping (Result<StorageListResult, Error>) -> Void) {
+    listAll(completion: getResultCallback(completion: completion))
+  }
+
+  func putData(uploadData: Data,
+               metadata: StorageMetadata?,
+               completion: @escaping (Result<StorageMetadata, Error>) -> Void)
+    -> StorageUploadTask {
+    return putData(uploadData,
+                   metadata: metadata,
+                   completion: getResultCallback(completion: completion))
+  }
+
+  func putFile(from: URL,
+               metadata: StorageMetadata?,
+               completion: @escaping (Result<StorageMetadata, Error>) -> Void)
+    -> StorageUploadTask {
+    return putFile(from: from,
+                   metadata: metadata,
+                   completion: getResultCallback(completion: completion))
+  }
+
+  func updateMetadata(metadata: StorageMetadata,
+                      completion: @escaping (Result<StorageMetadata, Error>) -> Void) {
+    return updateMetadata(metadata, completion: getResultCallback(completion: completion))
+  }
+
+  func writeToFile(toFile: URL, completion: @escaping (Result<URL, Error>)
```

Comment:
`writeToFile(...)` is not the same as the underlying method. This should be:

```swift
func write(toFile: URL, ...);
```

Reply:
Done